src/main/java/com/google/devtools/build/lib/remote/RemoteActionInputFetcher.java

```diff
@@ -101,19 +106,16 @@ protected ListenableFuture<Void> doDownloadFile(
   protected Completable onErrorResumeNext(Throwable error) {
     if (error instanceof BulkTransferException) {
       if (((BulkTransferException) error).onlyCausedByCacheNotFoundException()) {
-        BulkTransferException bulkAnnotatedException = new BulkTransferException();
-        for (Throwable t : error.getSuppressed()) {
-          IOException annotatedException =
-              new IOException(
-                  String.format(
-                      "Failed to fetch file with hash '%s' because it does not"
```

Comment:
I know people use the hash from the error message for debugging. I also know they can look at the gRPC log to get the same hash, but not everyone collects that all the time. Are we able to preserve the hash in the error?

Reply:
The hash is saved in original `BulkTransferException` exception and will be printed out. e.g.

```
Failed to fetch blobs because they do not exist remotely. --remote_download_outputs=minimal does not work if your remote cache evicts blobs during builds: Missing digest: b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9/11
```

Updated test to assert on the hash.